internal/cmd/config_test.go

```diff
@@ -202,3 +206,199 @@ func TestDeriveAndValidateConfig(t *testing.T) {
+}
+
+func TestReadDiskConfigNotJSONContentError(t *testing.T) {
+	t.Parallel()
+	memfs := fsext.NewMemMapFs()
+
+	conf := []byte(`bad json format`)
+	defaultConfigPath := ".config/loadimpact/k6/config.json"
+	require.NoError(t, fsext.WriteFile(memfs, defaultConfigPath, conf, 0o644))
+
+	gs := &state.GlobalState{
+		FS:    memfs,
+		Flags: state.GetDefaultFlags(".config"),
+	}
+	_, err := readDiskConfig(gs)
+	require.ErrorContains(t, err, "couldn't parse")
+}
+
+func TestReadDiskConfigNotFoundErrorWithCustomPath(t *testing.T) {
+	t.Parallel()
+	memfs := fsext.NewMemMapFs()
+
+	defaultFlags := state.GetDefaultFlags(".config")
+	gs := &state.GlobalState{
+		FS:           memfs,
+		Flags:        defaultFlags,
+		DefaultFlags: defaultFlags,
+	}
+	gs.Flags.ConfigFilePath = ".config/my-custom-path/k6/config.json"
+
+	c, err := readDiskConfig(gs)
+	assert.ErrorIs(t, err, fs.ErrNotExist)
+	assert.Empty(t, c)
+}
+
+func TestWriteDiskConfigWithDefaultFlags(t *testing.T) {
+	t.Parallel()
+	memfs := fsext.NewMemMapFs()
+
+	defaultFlags := state.GetDefaultFlags(".config")
+	gs := &state.GlobalState{
+		FS:           memfs,
+		Flags:        defaultFlags,
+		DefaultFlags: defaultFlags,
+	}
+
+	c := Config{WebDashboard: null.BoolFrom(true)}
+	err := writeDiskConfig(gs, c)
+	require.NoError(t, err)
+
+	finfo, err := memfs.Stat(".config/loadimpact/k6/config.json")
+	require.NoError(t, err)
+	assert.NotEmpty(t, finfo.Size())
+}
+
+func TestWriteDiskConfigOverwrite(t *testing.T) {
+	t.Parallel()
+	memfs := fsext.NewMemMapFs()
+
+	conf := []byte(`{"iterations":1028,"cloud":{"field1":"testvalue"}}`)
+	defaultConfigPath := ".config/loadimpact/k6/config.json"
+	require.NoError(t, fsext.WriteFile(memfs, defaultConfigPath, conf, 0o644))
+
+	defaultFlags := state.GetDefaultFlags(".config")
+	gs := &state.GlobalState{
+		FS:           memfs,
+		Flags:        defaultFlags,
+		DefaultFlags: defaultFlags,
+	}
+
+	c := Config{WebDashboard: null.BoolFrom(true)}
+	err := writeDiskConfig(gs, c)
+	require.NoError(t, err)
+}
+
+func TestWriteDiskConfigCustomPath(t *testing.T) {
+	t.Parallel()
+	memfs := fsext.NewMemMapFs()
+
+	defaultFlags := state.GetDefaultFlags(".config")
+	gs := &state.GlobalState{
+		FS:           memfs,
+		Flags:        defaultFlags,
+		DefaultFlags: defaultFlags,
+	}
+	gs.Flags.ConfigFilePath = "my-custom-path/config.json"
+
+	c := Config{WebDashboard: null.BoolFrom(true)}
+	err := writeDiskConfig(gs, c)
+	require.NoError(t, err)
+}
+
+func TestWriteDiskConfigNoJSONContentError(t *testing.T) {
+	t.Parallel()
+	memfs := fsext.NewMemMapFs()
+
+	defaultFlags := state.GetDefaultFlags(".config")
+	gs := &state.GlobalState{
+		FS:           memfs,
+		Flags:        defaultFlags,
+		DefaultFlags: defaultFlags,
+	}
+
+	c := Config{
+		WebDashboard: null.BoolFrom(true),
+		Options: lib.Options{
+			Cloud: []byte(`invalid-json`),
+		},
+	}
+	err := writeDiskConfig(gs, c)
+	assert.ErrorContains(t, err, "json: error")
```

Comment:
Same suggestion from before:

```suggestion
	var serr *json.SyntaxError
	assert.ErrorAs(t, err, &jerr)
```